scripts/trivydb2tc.py

```diff
@@ -470,102 +427,99 @@ def main() -> None:
-                for vuln in vulns:
+                trivy_vulns = vuln_info(repos, txs)
+                for vuln in trivy_vulns:
                     solution, vuln_vers, fix_vers = solution_from_vuln(vuln)
                     if not solution:
                         continue
-                    vuln_id = vuln["vuln_id"]
-                    vuln_obj = vuln_dict.get(
-                        vuln_id, {"tags": set(), "actions": {}, "fix_vers": {}}
-                    )
-                    tag = f"{vuln['pkg_name']}:{category}"
-                    assert isinstance(vuln_obj["tags"], set)
-                    vuln_obj["tags"].add(tag)
-                    assert isinstance(vuln_obj["actions"], dict)
-                    act_obj = vuln_obj["actions"].get(
+                    trivy_vuln_id = vuln["vuln_id"]
+                    vuln_obj = vuln_dict.get(trivy_vuln_id, {"affects": {}})
+                    package = f"{vuln['pkg_name']}:{category}"
+
+                    # Ensure "affects" is dict
+                    assert isinstance(vuln_obj["affects"], dict)
+                    affect_obj = vuln_obj["affects"].get(
                         solution,
                         {
-                            "action": solution,
-                            "action_type": "elimination",
-                            "recommended": True,
-                            "ext": {
-                                "tags": set(),
-                                "vulnerable_versions": {},  # {tag: versions}
-                            },
+                            "packages": set(),
+                            "affected_versions": set(),
+                            "fixed_versions": set(),
                         },
                     )
-                    act_obj["ext"]["tags"].add(tag)
-                    vers_obj = act_obj["ext"]["vulnerable_versions"].get(tag, set())
-                    vers_obj.update(vuln_vers or [])
-                    act_obj["ext"]["vulnerable_versions"][tag] = vers_obj
-                    vuln_obj["actions"][solution] = act_obj
-                    vuln_obj["fix_vers"] = fix_vers
-                    vuln_dict[vuln_id] = vuln_obj
-
-        topics: dict[str, dict] = {}
+
+                    # Add the package to the "affects" set
+                    affect_obj["packages"].add(package)
+
+                    # Update "affected_versions"
+                    affect_vers_obj = affect_obj["affected_versions"]
+                    affect_vers_obj.update(vuln_vers or [])
+                    affect_obj["affected_versions"] = affect_vers_obj
+
+                    # Update "fixed_versions"
+                    fixed_vers_obj = affect_obj["fixed_versions"]
+                    fixed_vers_obj.update(fix_vers or [])
+                    affect_obj["fixed_versions"] = fixed_vers_obj
+
+                    # Update the "vuln_dict"
+                    vuln_obj["affects"][solution] = affect_obj
```

Comment:
solutionでなくpackageで集約すべき
```suggestion
                    vuln_obj["affects"][package] = affect_obj
```
affect_objに現状"packages"をキーとしてpackageを持つが、これは不要になる。
solutionは不要になるので、solution_from_vuln()も必要な情報だけ返すよう見直し必要。

変更の根拠として、affected_versionsはAffect毎に集約する。
Affectはpackage_idを持つため、Packageテーブルのキーである{name, ecosystem}を組み合わせたpackageをキーにして集約すると良い。